lib/mix/lib/mix/tasks/release.init.ex

```diff
@@ -87,8 +87,20 @@ defmodule Mix.Tasks.Release.Init do
     #!/bin/sh
     set -e
 
-    SELF=$(readlink "$0" || true)
-    if [ -z "$SELF" ]; then SELF="$0"; fi
+    readlink_f () {
+      original_dir=$(pwd)
+      cd "$(dirname "$1")" > /dev/null
+      filename="$(basename "$1")"
+      if [ -h "$filename" ]; then
+        result=$(readlink_f "$(readlink "$filename")")
+      else
+        result="$(pwd -P)/$filename"
+      fi
+      cd "$original_dir" > /dev/null
+      echo "$result"
+    }
```

Comment:
is there a reason wy we are not using the same readlink_f as bin/elixir? Wouldn't it be better to standardize? Thank you!

Reply:
I left a note on your previous comment about this.

The `readlink_f` function in `bin/elixir` changes the current directory in order to resolve the relative paths correctly, so it ends in a different directory than where it started.

It seems to me that the current implementation of both `cli_text` and `cli_bat_text` in `release.init` preserves the original directory. The .bat version uses `pushd` and `popd` to achieve this:

```
pushd .
cd "%~dp0\.."
set RELEASE_ROOT=%cd%
popd
```

I did that in order to preserves the same behavior, but I'm not sure if there is a real side-effect on changing the current directory in this scenario.

Reply:
I am not sure the changing of the directory that happens there affects the actual program, does it? After all, I can run `elixir -e "IO.inspect File.cwd"` from anywhere, and it will preserve it. So if we do it, we are either reverting back somewhere (or sh is doing it), no?

Reply:
Yes, it does work fine with original `readlink_f` implementation.

I tested reading a file from a path relative to the current directory using a relative path symlink, and it worked as expected:

```
bin/test_release_relative eval "TestReleaseProject.read_file(\"../README.md\")"
```

The bat version handled returning to the original directory, which made me think there might be more going on behind the scenes.

I’ve reverted to the original implementation since it maintains the expected behavior without unnecessary complexity.